polaris-react/src/components/DataTable/DataTable.tsx

```diff
@@ -477,13 +501,6 @@ class DataTableInner extends PureComponent<CombinedProps, DataTableState> {
             ? currentColumn.rightEdge
             : previousColumn.leftEdge;
 
-        if (stickyTableHeadingsRow) {
```

Comment:
Removed this as we don't need to specific the moving of the sticky heading, the table will do it for us.

**Before**: 
Navigation button is clicked 
-> JS scrolls `stickyTableHeadingsRow`

**After**: 
Navigation button is clicked 
-> JS scrolls `scrollContainer` 
-> `scrollContainer` scrolls `stickyTableHeadingsRow` (because of `private stickyHeaderScrolling`)